common/expression/src/column_from.rs

```diff
@@ -0,0 +1,128 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::iter::Iterator;
+
+use crate::types::nullable::NullableColumn;
+use crate::types::number::*;
+use crate::types::*;
+use crate::Column;
+
+/// ColumnFrom is a helper trait to generate columns.
+pub trait ColumnFrom<D, Phantom: ?Sized> {
+    /// Initialize by name and values.
+    fn from_data(_: D) -> Column;
+
+    fn from_data_valids(d: D, valids: Vec<bool>) -> Column {
+        let column = Self::from_data(d);
```

Comment:
Will we have an assert here to make sure `d` has the same length with `valids`?

BTW, for better naming:

- `from_data_valids` -> `from_data_with_validity`?
- `valids` -> `validity`?

Reply:
I considered that ,but feel`from_data_with_validity ` is too long to write.

Reply:
But easier for reading. I think it's a good tradeoff.

Reply:
Ok